nikola/plugins/command/import_wordpress.py

```diff
@@ -536,17 +601,45 @@ def process_item(self, item):
         wordpress_namespace = item.nsmap['wp']
         post_type = get_text_tag(
             item, '{{{0}}}post_type'.format(wordpress_namespace), 'post')
+        post_id = int(get_text_tag(
+            item, '{{{0}}}post_id'.format(wordpress_namespace), "0"))
+        parent_id = get_text_tag(
+            item, '{{{0}}}post_parent'.format(wordpress_namespace), None)
 
         if post_type == 'attachment':
-            self.import_attachment(item, wordpress_namespace)
-        elif post_type == 'post':
-            self.import_item(item, wordpress_namespace, 'posts')
+            files = self.import_attachment(item, wordpress_namespace)
+            # If parent was found, store relation with imported files
+            if parent_id is not None:
+                self.attachments[int(parent_id)][post_id] = files
+            else:
+                LOGGER.warn("Attachment #{0} ({1}) has no parent!".format(post_id, files))
         else:
-            self.import_item(item, wordpress_namespace, 'stories')
+            if post_type == 'post':
+                out_folder_slug = self.import_item(item, wordpress_namespace, 'posts')
+            else:
+                post_type = 'page'
+                out_folder_slug = self.import_item(item, wordpress_namespace, 'stories')
+            # If post was exported, store data
+            if out_folder_slug:
+                self.posts_pages[post_id] = (post_type, out_folder_slug[0], out_folder_slug[1])
+
+    def write_attachments_info(self, path, attachments):
```

Comment:
These files could be used by plugins for the WordPress page compiler. For example, if you include attachments into a post as a gallery, you have something like
```[gallery columns="3" link="file" ids="52,53,54,55,56,57,58"]```
inside the post -- here 52 up to 58 are IDs for attachments. If you have this `.json` file, you can map the IDs to actual images.